### DISCUSSION TITLE
API call with attachment field [Sep 26]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#3922] [Discord Threads]
> 🙋‍♂️ **Guy Butts (datalytyks)**

**Comment by o1lab** [comment]
> 👨‍🦲 **dstala**  
#### use storage API : `upload` or `uploadByUrl` to add attachment file into storage;  
#  
> 👨‍💻 **pranavxc**  
#### 1. First you have to upload file using our storage api - https://all-apis.nocodb.com/#tag/Storage/operation/storage-upload   
#### 2. Use the response from storage api and provide the attachment column as JSON array of attachment( response from first one )   
####  - https://all-apis.nocodb.com/#tag/DB-table-row/operation/db-table-row-create   
####    
#### Sample payload ```json   
#### {   
####    "title": "abc",   
####    "attachmenttColumnName": "[{\"url\": \"string\", \"fileName\":\"string\", \"mimetype\": \"string\", \"size\": \"string\"}]"   
#### }   
#### ```  
#  
> 🙋‍♂️ **Guy Butts (datalytyks)**  
#### Beautiful, works like a charm and I’m back in business, thanks!! Loving NocoDB so far!  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/1021094701522239550)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
